packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,217 @@
+const _ = require('lodash');
+const path = require('path');
+const colors = require('colors/safe');
+
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+const utils = require('../../utils');
+
+class ScaffoldCommand extends BaseCommand {
+  writeTemplateFile(templatePath, templateContext, dest) {
+    const destPath = path.join(process.cwd(), `${dest}.js`);
+    const preventOverwrite = !this.flags.force;
+
+    if (preventOverwrite && utils.fileExistsSync(destPath)) {
+      const [location, filename] = dest.concat('.js').split('/');
+      return Promise.reject(
+        [
+          `File ${colors.bold(filename)} already exists within ${colors.bold(
+            location
+          )}.`,
+          'You could:',
+          '1. Choose a different filename',
+          `2. Delete ${filename} from ${location}`,
+          `3. Run ${colors.italic('scaffold')} with ${colors.bold(
+            '--force'
+          )} to overwrite the current ${filename}`
+        ].join('\n')
+      );
+    }
+
+    return utils
+      .readFile(templatePath)
+      .then(templateBuf => templateBuf.toString())
+      .then(template =>
+        _.template(template, { interpolate: /<%=([\s\S]+?)%>/g })(
+          templateContext
+        )
+      )
+      .then(rendered => {
+        this.startSpinner(`Writing new ${dest}.js`);
+        return utils
+          .ensureDir(path.dirname(destPath))
+          .then(() => utils.writeFile(destPath, rendered));
+      })
+      .then(this.stopSpinner);
+  }
+
+  async perform() {
+    const { name, type } = this.args;
+    const templateContext = {
+      CAMEL: utils.camelCase(name),
+      KEY: utils.snakeCase(name),
+      NOUN: _.capitalize(name),
+      LOWER_NOUN: name.toLowerCase(),
+      INPUT_FIELDS: '',
+      TYPE: type,
+      TYPE_PLURAL: type === 'search' ? `${type}es` : `${type}s`,
+      // resources need an extra line for tests to "just run"
+      MAYBE_RESOURCE: type === 'resource' ? 'list.' : ''
+    };
+
+    // what is the `resources: {}` app definition point?
+    const typeMap = {
+      resource: 'resources',
+      trigger: 'triggers',
+      search: 'searches',
+      create: 'creates'
+    };
+
+    // where will we create/required the new file?
+    const destMap = {
+      resource: `resources/${templateContext.KEY}`,
+      trigger: `triggers/${templateContext.KEY}`,
+      search: `searches/${templateContext.KEY}`,
+      create: `creates/${templateContext.KEY}`
+    };
+
+    if (!typeMap[type]) {
+      this.log(
```

Comment:
this could be a `this.error`, which will throw instead of returning a "positive" response.